packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -2238,13 +2269,30 @@ export class Bucket extends BucketBase {
    * @param props Par
    */
   private parseLifecycleConfiguration(): CfnBucket.LifecycleConfigurationProperty | undefined {
+    const isValidTransitionDefaultMinimumObjectSize = this.lifecycleRules.every((rule: LifecycleRule): boolean =>
+      rule.abortIncompleteMultipartUploadAfter !== undefined ||
+      rule.expiration !== undefined ||
+      rule.expirationDate !== undefined ||
+      rule.expiredObjectDeleteMarker !== undefined ||
+      rule.noncurrentVersionExpiration !== undefined ||
+      rule.noncurrentVersionsToRetain !== undefined ||
+      rule.noncurrentVersionTransitions !== undefined ||
+      rule.transitions !== undefined
+    );
+    if (this.transitionDefaultMinimumObjectSize && !isValidTransitionDefaultMinimumObjectSize) {
+      throw new Error('TransitionDefaultMinimumObjectSize cannot be specified if all lifecycleRules don\'t have at least one of the following properties: abortIncompleteMultipartUploadAfter, expiration, expirationDate, expiredObjectDeleteMarker, noncurrentVersionExpiration, noncurrentVersionsToRetain, noncurrentVersionTransitions, or transitions');
+    }
```

Comment:
Otherwise a CFn error occurs.

> Invalid request provided: At least one of [ExpirationDate,ExpirationInDays,AbortIncompleteMultipartUpload,Transition,Transitions,NoncurrentVersionExpirationInDays,NoncurrentVersionTransition,NoncurrentVersionTransitions,NoncurrentVersionExpiration,ExpiredObjectDeleteMarker] needs to be specified